pkg/controller/performanceprofile/components/utils.go

```diff
@@ -26,3 +28,65 @@ func SplitLabelKey(s string) (domain, role string, err error) {
 	}
 	return parts[0], parts[1], nil
 }
+
+func cpusListToArray(cpusListStr string) ([]int, error) {
+	var cpusList []int
+	elements := strings.Split(cpusListStr, ",")
+	for _, item := range elements {
+		cpuRange := strings.Split(item, "-")
+		// provided a range: 1-3
+		if len(cpuRange) > 1 {
+			start, err := strconv.Atoi(cpuRange[0])
+			if err != nil {
+				return nil, err
+			}
+			end, err := strconv.Atoi(cpuRange[1])
+			if err != nil {
+				return nil, err
+			}
+			// Add cpus to the list. Assuming it's a valid range.
+			for cpuNum := start; cpuNum <= end; cpuNum++ {
+				cpusList = append(cpusList, cpuNum)
+			}
+		} else {
+			cpuNum, err := strconv.Atoi(cpuRange[0])
+			if err != nil {
+				return nil, err
+			}
+			cpusList = append(cpusList, cpuNum)
+		}
+	}
+
+	return cpusList, nil
+}
+
+// CPUListToHexMask converts a list of cpus into a cpu mask represented in hexdecimal
+func CPUListToHexMask(cpulist string) (hexMask string) {
+	reservedCpus, _ := cpusListToArray(cpulist)
+	currMask := big.NewInt(0)
+	for _, cpu := range reservedCpus {
+		x := new(big.Int).Lsh(big.NewInt(1), uint(cpu))
+		currMask.Or(currMask, x)
+	}
+	return fmt.Sprintf("%02x", currMask)
+}
+
+// CPUListToInvertedMask converts a list of cpus into an inverted cpu mask represented in hexdecimal
+func CPUListToInvertedMask(cpulist string) (hexMask string) {
+	reservedCpus, _ := cpusListToArray(cpulist)
+
+	reservedCpusLookup := make(map[int]bool)
+	for _, cpu := range reservedCpus {
+		reservedCpusLookup[cpu] = true
+	}
+
+	currMask := big.NewInt(0)
+	for cpu := 0; cpu < 254; cpu++ {
```

Comment:
If you're going to respin this or add new commits, please make this a const (MAX_SYSTEM_CPUS or something ), I forgot about it.